### PR TITLE
Add rust-boon implementation

### PR DESCRIPTION
**Pull request by santhosh-tekuri**
https://crates.io/crates/boon

### CODE REVIEW

File: implementations/rust-boon/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+RUN update-ca-certificates
+
+WORKDIR /usr/src/myapp
+COPY . .
+
+ENV RUSTFLAGS='-C linker=x86_64-linux-gnu-gcc'
+RUN cargo build --target x86_64-unknown-linux-musl --release
+
+FROM alpine
+COPY --from=builder /usr/src/myapp/target/x86_64-unknown-linux-musl/release/bowtie-rust-jsonschema /usr/local/bin
```
Comment (Julian) [inline implementations/rust-boon/Dockerfile:14]:
```suggestion
COPY --from=builder /usr/src/myapp/target/x86_64-unknown-linux-musl/release/bowtie-rust-boon /usr/local/bin
```

File: implementations/rust-boon/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+[package]
+name = "bowtie-rust-boon"
+version = "0.1.0"
+edition = "2021"
+
+[dependencies]
```
Comment (Julian) [inline implementations/rust-boon/Cargo.toml:6]: Can you add this directory to .github/dependabot so it's kept autoupdated? It should go in 2 places, both to keep the `Dockerfile` updated and also your lib.

And also, can you add the directory to `.pre-commit-config.yaml` which should just ensure that `cargo fmt` is run on the rust file? You'll see `rust-jsonschema` already in there.

**Review by Julian — CHANGES_REQUESTED** [review]
Nice, thanks!

It seems CI isn't running the image building job on PRs for some reason, I'll have to look into why and fix that, but this won't build yet for a minor reason which I left a comment on, just a small copypaste typo.

Other than that and the dependencies stuff looks great though, can you address those and then happy to merge, appreciated!

File: implementations/rust-boon/src/main.rs [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+                started = true;
+                if request["version"] != 1 {
+                    Err("not version 1")?;
+                }
+                let response = json!({
+                    "ready": true,
+                    "version": 1,
+                    "implementation": {
+                        "language": "rust",
+                        "name": "jsonschema",
```
Comment (Julian) [inline implementations/rust-boon/src/main.rs:40]:
```suggestion
                        "name": "boon",
```

File: implementations/rust-boon/Dockerfile (hunk 1 above)
Comment (santhosh-tekuri) [reply implementations/rust-boon/Dockerfile]: done

File: implementations/rust-boon/Cargo.toml (hunk 2 above)
Comment (santhosh-tekuri) [reply implementations/rust-boon/Cargo.toml:6]: done

File: implementations/rust-boon/src/main.rs (hunk 3 above)
Comment (santhosh-tekuri) [reply implementations/rust-boon/src/main.rs]: done

**Pull request merged**